### DISCUSSION TITLE
Message Persistence v5.6.1 (OpenSource) - Question

### DISCUSSION THREAD

**Discussion by ennc0d3** [emqx/emqx#13117] [Q&A] [answered]
I tried with following EMQX configuration, in v5.6.1 (OpenSource)

```
    durable_storage {
        messages {
           backend = "builtin"
        }
      }

```
When I check the the configuration using `emqx ctl ds`, it says the ds is *disabled*. Does this mean the durable storage for message persistence is not supported ?

```
emqx@emqx-core-99ddfbd86-0:/opt/emqx$ emqx ctl conf show durable_storage
durable_storage {
  messages {backend = builtin}
}

emqx@emqx-core-99ddfbd86-0:/opt/emqx$ emqx ctl ds
ds # Durable storage is disabled
```

Still,  I checked by publishing a message, restarted the core node ( Only 1 core in my case) and then subscribed to the topic, I didn't the message.

I checked an similar issue, https://github.com/emqx/emqx/discussions/12573 which tates that the feature is in for testing, but doesn't seem like working with the configuration. 

Could you please let me know the status ?

**Comment by ieQu1** [comment]
Hello,

Durable sessions in 5.6 release were in developer preview state. Please wait for 5.7.0 (should be released next Monday, fingers crossed). It will come with documentation and a lot of bugfixes.

To answer your question, `durable_storage` is a config subtree that configures _how_ messages are stored, but not _whether_ they are stored or not. The latter option is enabled via `durable_sessions.enable = true` parameter (on 5.7). It's `false` by default, so EMQX doesn't create the durable storage for messages. `durable_sessions` feature is totally dormant unless enabled explicitly, so users who don't need it can keep using EMQX as usual.

**Comment by ennc0d3** [comment]
Cool, Thanks. If I understand correctly, the durable_sessions feature can be only enabled in 5.7 and not in preview mode?

**Reply by ieQu1** [reply, marked as answer]
It can be enabled in 5.6. You probably don't want this, though, because you will miss a couple dozens of bugfixes. We're preparing 5.7 release cut as we speak.

**Comment by ieQu1** [comment]
This document explains how to activate the feature on EMQX 5.7

https://www.emqx.io/docs/en/latest/durability/durability_introduction.html

**Reply by martinzima** [reply]
This page is gone now (404). What happened? :(
